injects/logger.js

```diff
@@ -66,7 +66,9 @@ hook.enable()
 // before process exits, flush the encoded data to the sample file
 process.once('beforeExit', function () {
   hook.disable()
-  encoder.end()
+  encoder.end(() => {
+    out.destroy()
```

Comment:
Can you please add a comment describing the problem a `TODO`? Then let's land this.